test/docker/Dockerfile

```diff
@@ -0,0 +1,134 @@
+FROM ubuntu:22.04
+
+
+# defaults
+RUN \
```

Comment:
Too many compiled sources from other repos, it will be hard to maintain, Is there an official Docker image for NetOpeer

Reply:
@kuisathaverat no there is no official docker image. I cannot see any other way to do it. What do you suggest?

Reply:
I do not feel comfortable adding 6min build from code five libs and one binary

Reply:
I pushed the Docker image to this repository as `ghcr.io/jenkinsci/trilead-ssh2:netopeer-3.5.1`, so we will keep the Dockerfile, but we will use a compiled Docker image. I saw the Docker image is huge, 1.6GB; maybe we can split it into two stages or remove compilation stuff

Reply:
@kuisathaverat so no need to change the testcontainer code? Yes I agree it is large but could not really find something that supports ssh call home. Any optimization is fine for me. 
Do you expect anything more from me?

Reply:
the image is pushed to the repo so no action needed on your end.

Reply:
@kuisathaverat  I changed to run with the image in testcontainers.
Oct 22, 2024 7:08:15 AM com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.Wire wire
FINE: http-outgoing-2 << "{"message":"Head \"https://ghcr.io/v2/jenkinsci/trilead-ssh2/manifests/netopeer-3.5.1\": unauthorized"}[\n]"

Reply:
@kuisathaverat do I need special access rights?

Reply:
I forget I have to make it public 

https://github.com/jenkinsci/trilead-ssh2/pkgs/container/trilead-ssh2

Reply:
I also tried this locally:

```
$ docker build -t ghcr.io/jenkinsci/trilead-ssh2:netopeer-3.5.1 .
[+] Building 0.8s (2/2) FINISHED                                                                                                                                                                                                                            docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                  0.6s
 => => transferring dockerfile: 2B                                                                                                                                                                                                                                    0.3s 
 => [internal] load .dockerignore                                                                                                                                                                                                                                     0.6s
 => => transferring context: 2B                                                                                                                                                                                                                                       0.3s 
ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount4081330339/Dockerfile: no such file or directory
```

Reply:
you need to change to the folder `test/docker` first

Reply:
@kuisathaverat  When we use testcontainers we should be able to pull the image. However I cannot access the image:

```
FINE: http-outgoing-2 << "{"message":"Head "[https://ghcr.io/v2/jenkinsci/trilead-ssh2/manifests/netopeer-3.5.1\](https://ghcr.io/v2/jenkinsci/trilead-ssh2/manifests/netopeer-3.5.1%5C)": unauthorized"}[\n]"
```

Reply:
It works on my local and in the CI